model_templates/triton_onnx_unstructured/client/datarobot-predict.py

```diff
@@ -0,0 +1,147 @@
+
+
+class DataRobotPredictionError(Exception):
+    """Raised if there are issues getting predictions from DataRobot"""
+
+
+def make_datarobot_deployment_unstructured_predictions(data, deployment_id, mimetype, charset):
+    """
+    Make unstructured predictions on data provided using DataRobot deployment_id provided.
+    See docs for details:
+         https://docs.datarobot.com/en/docs/api/reference/predapi/dr-predapi.html
+ 
+    Parameters
+    ----------
+    data : bytes
+        Bytes data read from provided file.
+    deployment_id : str
+        The ID of the deployment to make predictions with.
+    mimetype : str
+        Mimetype describing data being sent.
+        If mimetype starts with 'text/' or equal to 'application/json',
+        data will be decoded with provided or default(UTF-8) charset
+        and passed into the 'score_unstructured' hook implemented in custom.py provided with the model.
+ 
+        In case of other mimetype values data is treated as binary and passed without decoding.
+    charset : str
+        Charset should match the contents of the file, if file is text.
+ 
+    Returns
+    -------
+    data : bytes
+        Arbitrary data returned by unstructured model.
+ 
+ 
+    Raises
+    ------
+    DataRobotPredictionError if there are issues getting predictions from DataRobot
+    """
+    # Set HTTP headers. The charset should match the contents of the file.
+    headers = {
+        'Content-Type': '{};charset={}'.format(mimetype, charset),
+        'Authorization': 'Bearer {}'.format(API_KEY),
+        'DataRobot-Key': DATAROBOT_KEY,
+    }
+
+    url = API_URL.format(deployment_id=deployment_id)
+
+    # Make API request for predictions
+    predictions_response = requests.post(
+        url,
+        data=data,
+        headers=headers,
+    )
+    _raise_dataroboterror_for_status(predictions_response)
+    # Return raw response content
+    return predictions_response.content
+
+
+def _raise_dataroboterror_for_status(response):
+    """Raise DataRobotPredictionError if the request fails along with the response returned"""
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        err_msg = '{code} Error: {msg}'.format(
+            code=response.status_code, msg=response.text)
+        raise DataRobotPredictionError(err_msg)
+
+
+def preprocess_image(img_path):
+    img = Image.open(img_path)
+    preprocess = transforms.Compose(
+        [
+            transforms.Resize(256),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+        ]
+    )
+    return np.expand_dims(preprocess(img).numpy(), axis=0)
+
+
+def read_model_inference_header():
+    with open('model_header.json') as f:
+        json_header = json.load(f)
+        json_header_str = json.dumps(json_header) + '\n'
+    return json_header_str.encode('utf-8')
```

Comment:
I'd put a comment in here stating that the JSON output needs to be minified to a single line.

Reply:
I'm going to simplify this, using the cusom.py hooks. So users don't need to know these details.

Reply:
And if a user wants to use their own `custom.py`? I presume it will be impossible to keep this detail completely hidden from the user?